src/common/messaging/icpc_messaging_component.ts

```diff
@@ -0,0 +1,237 @@
+import { v4 as uuidv4 } from 'uuid';
```

Comment:
copyright. also what is ICPC from reading the code alone? needs to be nicely documented